tabular/src/autogluon/tabular/models/text_prediction/text_prediction_v1_model.py

```diff
@@ -0,0 +1,270 @@
+            from autogluon.text.text_prediction.dataset import TabularDataset
+        except ImportError:
+            raise ImportError(AG_TEXT_IMPORT_ERROR)
+
+        # Get arguments from kwargs
+        verbosity = kwargs.get('verbosity', 2)
+        num_cpus = kwargs.get('num_cpus')
+        num_gpus = kwargs.get('num_gpus')
+        X_train = X_train[['sentence_raw_text']]
+        X_val = X_val[['sentence_raw_text']]
+        column_properties = self._build_model(X_train=X_train,
+                                              y_train=y_train,
+                                              X_val=X_val,
+                                              y_val=y_val,
+                                              hyperparameters=self.params)
+        print('X_train=', X_train)
+        # Insert the label column
+        X_train.insert(len(X_train.columns), self._label_column_name, y_train)
+        X_val.insert(len(X_val.columns), self._label_column_name, y_val)
+        scheduler_options = self.params['hpo_params']['scheduler_options']
+        search_strategy = self.params['hpo_params']['search_strategy']
+        if scheduler_options is None:
+            scheduler_options = dict()
+        if search_strategy.endswith('hyperband'):
+            # Specific defaults for hyperband scheduling
+            scheduler_options['reduction_factor'] = scheduler_options.get(
+                'reduction_factor', 4)
+            scheduler_options['grace_period'] = scheduler_options.get(
+                'grace_period', 10)
+            scheduler_options['max_t'] = scheduler_options.get(
+                'max_t', 50)
+        train_data = TabularDataset(X_train,
+                                    column_properties=column_properties,
+                                    label_columns=self._label_column_name)
+        tuning_data = TabularDataset(X_val,
+                                     column_properties=column_properties,
+                                     label_columns=self._label_column_name)
+        self.model.train(train_data=train_data,
+                         tuning_data=tuning_data,
+                         resource={'num_cpus': num_cpus,
+                                   'num_gpus': num_gpus},
+                         time_limits=time_limit,
+                         search_strategy=search_strategy,
+                         search_options=self.params['hpo_params']['search_options'],
+                         scheduler_options=scheduler_options,
+                         num_trials=self.params['hpo_params']['num_trials'],
+                         console_log=verbosity >= 2,
+                         ignore_warning=verbosity < 2)
+
+    def save(self, path: str = None, verbose=True) -> str:
+        if path is None:
+            path = self.path
+        logger.log(15, f'Save to {path}.')
+        self.model.save(path)
+        return path
+
+    @classmethod
+    def load(cls, path: str, reset_paths=True, verbose=True):
```

Comment:
Missing `super().load` call, this does not return the correct object. load should load the `TextPredictionV1Model` object.

Reply:
I've revised to build the object based on the loaded model_info and the model.

Reply:
I think it will be more maintainable if we save this object as a pkl instead of reconstructing upon load.

See this code snippet taken from `NNFastAiTabularModel` :

```
    def save(self, path: str = None, verbose=True) -> str:
        __model = self.model
        self.model = None
        path_final = super().save(path=path, verbose=verbose)
        self.model = __model

        # Export model
        save_pkl.save_with_fn(f'{path_final}{self.model_internals_file_name}', self.model, lambda m, buffer: m.export(buffer), verbose=verbose)
        return path_final

    @classmethod
    def load(cls, path: str, reset_paths=True, verbose=True):
        from fastai.basic_train import load_learner
        model = super().load(path, reset_paths=reset_paths, verbose=verbose)
        model.model = load_pkl.load_with_fn(f'{model.path}{model.model_internals_file_name}', lambda p: load_learner(model.path, p), verbose=verbose)
        return model
```